compiler/hash-ast/src/ast.rs

```diff
@@ -13,7 +13,13 @@ use hash_source::{
     location::Span,
 };
 use hash_tree_def::define_tree;
-use hash_utils::{counter, smallvec::SmallVec};
+use hash_utils::{
+    counter,
+    index_vec::{Idx, IndexVec},
+    parking_lot::RwLock,
+    smallvec::SmallVec,
+};
+use once_cell::sync::Lazy;
 use replace_with::replace_with_or_abort;
 
 counter! {
```

Comment:
Since you are using indexvec then does it make sense to create this ID through https://doc.rust-lang.org/beta/nightly-rustc/rustc_index/macro.newtype_index.html?

Reply:
Yeah I was thinking about doing this, probably don’t need the `counter!` anymore.

Reply:
Will this retain the invariant of increasing `AstNodeIds` across the AST for ordering error messages? Also maybe that should be in a comment somewhere here.

Reply:
Yes, it will because only `AstNode::new()` can create a new AstNodeId and if they are made in a particular order, then that order should be preserved (as shown by the tests passing which rely on this behaviour to be true).